### PR TITLE
Chore: Restructure test_app

### PR DESCRIPTION
**Pull request by peter-marcisovsky**
The original `~/test_app` folder containing all the test cases was removed. Now each folder from `~/host/class` and `~/device` has its own `test_app` with its own `pytest_xxx_.py`

Proposed structure with separate test apps:

tests with `usb_host` tag: 
- `pytest_usb_host_cdc.py`
- `pytest_usb_host_hid.py`
- `pytest_usb_host_msc.py`
- `pytest_usb_host_uvc.py`

tests with `usb_device` tag:
- `pytest_usb_device_esp_tinyusb.py`
- `pytest_usb_device_cdc.py`

Tests with `usb_host` tag are run in CI
.
├── device
│   ├── dcd *
│   ├── esp_tinyusb
│   │   └── test_app
│   │      ├── main
│   │      ├── pytest_usb_device_esp_tinyusb.py
│   │      └── pytest_usb_device_cdc.py
│   └── power_delivery *
├── host
│   ├── class
│   │   ├── cdc
│   │   │   ├── esp_modem_usb_dte
│   │   │   ├── usb_host_cdc_acm
│   │   │      └── test_app
│   │   │         ├── main
│   │   │         └── pytest_usb_host_cdc.py
│   │   │   ├── usb_host_ch34x_vcp
│   │   │   ├── usb_host_cp210x_vcp
│   │   │   ├── usb_host_ftdi_vcp
│   │   │   └── usb_host_vcp
│   │   ├── hid
│   │   │   ├── esp_hid_usb *
│   │   │   └── usb_host_hid
│   │   │   └── test_app
│   │   │      ├── main
│   │   │      └── pytest_usb_host_hid.py
│   │   ├── msc
│   │   │   ├── usb_host_msc
│   │   │   └── test_app
│   │   │      ├── main
│   │   │      └── pytest_usb_host_msc.py
│   │   └── uvc
│   │       ├── usb_host_uvc
│   │       └── test_app
│   │          ├── main
│   │          └── pytest_usb_host_uvc.py
│   └── usb_host_lib *
├── otg *
└── usb_phy *

### CODE REVIEW

File: device/esp_tinyusb/test_app/main/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
-idf_component_register(SRCS "test_esp_tinyusb.c" "test_bvalid_sig.c"
+include($ENV{IDF_PATH}/tools/cmake/version.cmake)
+
+idf_component_register(SRCS "test_esp_tinyusb.c" "test_bvalid_sig.c" "test_app_main.c"
                        INCLUDE_DIRS "."
                        REQUIRES unity esp_tinyusb
-                       )
+                       WHOLE_ARCHIVE)
```
[the remark this answers is not in the document]
Comment (tore-espressif) [reply device/esp_tinyusb/test_app/main/CMakeLists.txt:6]: I see now, we do not test esp_tinyusb with IDF 4.4

File: device/esp_tinyusb/test_app/sdkconfig.defaults [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Configure TinyUSB, it will be used to mock USB devices
+CONFIG_TINYUSB=y
+CONFIG_TINYUSB_MSC_ENABLED=n
+CONFIG_TINYUSB_CDC_ENABLED=y
+CONFIG_TINYUSB_CDC_COUNT=2
+CONFIG_TINYUSB_HID_COUNT=0
+
+# Disable watchdogs, they'd get triggered during unity interactive menu
+CONFIG_ESP_INT_WDT=n
```
[the remark this answers is not in the document]
Comment (peter-marcisovsky) [reply device/esp_tinyusb/test_app/sdkconfig.defaults:9]: using `CONFIG_ESP_INT_WDT=n` in `sdkconfig.defaults` works in both, the latest and the IDF4.4

using `CONFIG_ESP_INT_WDT=n` result is the same for IDF4.4 and latest master:
1. You set `CONFIG_ESP_INT_WDT=n` in `sdkconfig.defaults` 
2. After building the project `sdkconfig` will say `# CONFIG_ESP_INT_WDT is not set`
3. `build/sdkconfig.h` does not have this config defined.

using `CONFIG_ESP_INT_WDT is not set` result is the same for IDF4.4 and latest master:
1. you specify `CONFIG_ESP_INT_WDT is not set` in `sdkconfig.defatuls`
2. After building the project `sdkconfig` will say `CONFIG_ESP_INT_WDT=y`
3. `build/sdkconfig.h` does have this config defined.

[the remark this answers is not in the document]
Comment (tore-espressif) [reply device/esp_tinyusb/test_app/sdkconfig.defaults:9]: Thank you for the verification!

**Review by tore-espressif — APPROVED** [review]
Thank you @peter-marcisovsky LGTM!

**Pull request merged**